### PR TITLE
SE-551-Update-Tron-to-support-Demos

### PR DESCRIPTION
**Pull request by ShaneDuncan602**
The Demo video can be found here: https://thracklelabs.atlassian.net/wiki/spaces/THRACKLELA/pages/284491794/Demos

### CODE REVIEW

File: script/DeployExampleGame.sh [hunk 1]
```diff
@@ -211,10 +216,12 @@ echo "################################################################"
 echo
 
 OUTPUT=$(forge create ./src/example/$APP_NAME/$FILE_NAME:"$APP_NAME"AppManager --constructor-args $APP_ADMIN_1 $APP_NAME false --private-key $APP_ADMIN_1_KEY --rpc-url $ETH_RPC_URL $GAS_ARGUMENT) 
-OUTPUTARRAY=$(echo $OUTPUT | tr '\n' ' ')
+INBETWEEN=$(echo $OUTPUT | sed 's/^.*Deployer/Deployer/')
```
Comment (mpetersoCode55) [inline script/DeployExampleGame.sh:217]: Someone sure bashed the heck out of this one

File: script/clientScripts/demo/Makefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Deploy contracts(singles)
+
+mintFranks:;	cast send $$APPLICATION_ERC721_1 "safeMint(address, uint256)" $USER_1 0 --private-key $APP_ADMIN_1_KEY --from $APP_ADMIN_1
```
Comment (mpetersoCode55) [inline script/clientScripts/demo/Makefile]: Should these be double dollar signs? 
You know what they say, mo money, mo problems

File: script/clientScripts/demo/test_env [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export APP_ADMIN_1=
```
Comment (mpetersoCode55) [inline script/clientScripts/demo/test_env]: can probably delete this, it's generated when the script runs purely from the parsed output

File: test_env [hunk 4]
```diff
@@ -0,0 +1,14 @@
+export APP_ADMIN_1=0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
```
Comment (mpetersoCode55) [inline test_env]: Can probably delete this, it's generated when the script runs purely from the parsed output

File: src/example/Frankenstein/FrankensteinAllowed.sol [hunk 5]
```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: UNLICENSED
```
Comment (mpetersoCode55) [inline src/example/Frankenstein/FrankensteinAllowed.sol]: Were these intended to be checked in? (I assume these were generated by the script)

File: .env [hunk 6]
```diff
@@ -1,5 +1,5 @@
 ## Protocol Contracts 
-RULE_PROCESSOR_DIAMOND=0x88dD21924FbEeccC4dC8Ae3a0e207ecCA03c5b92
+RULE_PROCESSOR_DIAMOND=0x0165878A594ca255338adfa4d48449f69242Eb8F
```
Comment (mpetersoCode55) [inline .env]: Did you intend to check in the change on the RULE_PROCESSOR_DIAMOND?

File: .env [hunk 7]
```diff
@@ -66,15 +66,15 @@ ACCESS_TIER_ADMIN=0x
 RISK_ADMIN=0x
 RULE_BYPASS_ACCOUNT=0x
 # commit to repo with 0x0000000000000000000000000000000000000000
-DEPLOYMENT_RULE_PROCESSOR_DIAMOND=0x0000000000000000000000000000000000000000
```
Comment (mpetersoCode55) [inline .env]: Same ? here

File: .env [hunk 8]
```diff
@@ -85,7 +85,7 @@ MUMBAI_APP_ADMIN_1=0x7E97c19CA80Ba38D64c8C2e047694a11459C23bB
 MUMBAI_APP_ADMIN_1_KEY=0x
 MUMBAI_APP_ADMIN_2=0x70040289973248574a5f91931170060EFba63aF8
 MUMBAI_APP_ADMIN_2_KEY=0x
-MUMBAI_RULE_PROCESSOR_DIAMOND=0x26DB256496aCc3cAC7F4D7254F8f96A947AFaaAf
```
Comment (mpetersoCode55) [inline .env:102]: same ? here

File: script/clientScripts/demo/DemoSetup.sh [hunk 9]
```diff
@@ -198,9 +198,9 @@ sed -i '' 's/ApplicationERC20Handler/'$APP_NAME'ERC20Handler/g' ./src/example/$A
```
Comment (mpetersoCode55) [inline script/clientScripts/demo/DemoSetup.sh]: Two thoughts:
1. Should we have duplicates of the setup logic in this script and the DemoSetup.sh script in the /script/demo directory?
2. This one is missing the fixes we did for the multiple compiler output issue

File: src/protocol/economic/ruleProcessor/ERC20TaggedRuleProcessorFacet.sol [hunk 10]
```diff
@@ -106,28 +106,31 @@ contract ERC20TaggedRuleProcessorFacet is IRuleProcessorErrors, IInputErrors, IT
...
-            for (uint i = 0; i < fromTags.length; ) {
-                TaggedRules.AccountMinMaxTokenBalance memory rule = getAccountMinMaxTokenBalance(ruleId, fromTags[i]);
-                /// check if period is 0, 0 means a period hasn't been applied to this rule
-                if(rule.period != 0) {
-                    /// Check to see if still in the hold period
-                    if ((block.timestamp - (uint256(rule.period) * 1 hours)) < startTime) {
-                        /// If the transaction will violate the rule, then revert
-                        if (rule.min > 0 && balanceFrom - amount < rule.min) revert TxnInFreezeWindow();
+        // if the balanceFrom is 0, then skip processing because this is a mint and it's impossible for a mint to violate the minium balance
+        if (balanceFrom!=0){
```
Comment (mpetersoCode55) [inline src/protocol/economic/ruleProcessor/ERC20TaggedRuleProcessorFacet.sol:110]: Nice!

**Review by mpetersoCode55 — CHANGES_REQUESTED** [review]

**Review by TJ-Everett — APPROVED** [review]
Yall are killin it with the scripting lately. Great Work Shane and I love the video. Super professional and hopefully saves a ton of time for you in the future

**Pull request merged**